torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -48,6 +48,20 @@ bool isValidReturnForRunning(Value* v) {
       v->type()->isSubtypeOf(NumberType::get());
 }
 
+bool containsTensorType(const TypePtr& t) {
+  if (t->isSubtypeOf(TensorType::get())) {
+    return true;
+  } else if (t->kind() == TypeKind::ListType) {
```

Comment:
This should use `virtual at::ArrayRef<TypePtr> containedTypes()` instead (defined on all types)  since it is missing cases for some types (optionals, futures).